contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -484,10 +486,11 @@ grant_guests_to_login(const char *login)
 
 		const char *db_name = TextDatumGetCString(db_name_datum);
 		char *guest_name = NULL;
+		AccessPriv *tmp = makeNode(AccessPriv);
+		
 		if (guest_role_exists_for_db(db_name))
-			guest_name = get_guest_role_name(db_name);
+			guest_name = (char *) get_guest_role_name(db_name);
```

Comment:
get_guest_role_name may return value which is palloc'd. So make sure we free it or make guest_name as const.